awx/api/metadata.py

```diff
@@ -169,6 +169,15 @@ def determine_actions(self, request, view):
                 # when reading a field and remove write-only fields.
                 if method == 'GET':
                     attrs_to_remove = ('required', 'read_only', 'default', 'min_length', 'max_length', 'placeholder')
+                    # Preserve any field data explicitly listed in Meta.preserve_metadata
+                    # Sometimes fields in attrs_to_remove (e.g. default) *are* actually important to show in GETs
+                    if hasattr(serializer, 'Meta') and hasattr(serializer.Meta, 'preserve_metadata'):
+                        preserve_metadata = serializer.Meta.preserve_metadata
+                        if field in preserve_metadata:
+                            if type(preserve_metadata[field]) is str:
+                                preserve_metadata[field] = {preserve_metadata[field]}
```

Comment:
Rather than mutating the value in `preserve_metadata`, I'd recommend killing these two lines and just making the specific example you are using `preserve_metadata = {'messages': ['default']}` or whatever.  If you'd still like to make things simpler in the case where there is only one piece of metadata to preserve, maybe just store the changed value in a variable instead of mutating.

Reply:
went back and forth on this - wasn't really sure if it made sense to include this part. Thanks for the feedback. Will drop this part and just send in a set in the first place. 👍